perf/src/sigverify.rs

```diff
@@ -271,6 +277,37 @@ fn do_get_packet_offsets(
         return Err(PacketError::InvalidPubkeyLen);
     }
 
+    // read instruction length
+    let instructions_len_offset = blockhash_offset
+        .checked_add(size_of::<Hash>())
+        .ok_or(PacketError::InvalidLen)?;
+
+    // Packet should have at least 1 more byte for instructions.len
+    let _ = instructions_len_offset
+        .checked_add(1usize)
+        .filter(|v| *v <= packet.meta().size)
+        .ok_or(PacketError::InvalidLen)?;
```

Comment:
Does this actually do anything?
`decode_shortu16_len` will return an `Err` if `packet.data(instructions_len_offset..)` is empty

Reply:
but that's basically the same as what `data()` does, right? it checks if discard() is true, then checks the bounds before returning Some or None. So if we leave this check ehre, we just check the bounds (well the upper bound) twice